sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -1,43 +1,457 @@
+/* global Slate */
 /**
  * TODO:
  * - move rendering responsibilities to the view?
  */
 Ext.define('SlateTasksTeacher.controller.Dashboard', {
     extend: 'Ext.app.Controller',
-
+    requires: [
+        'Ext.window.Toast'
+    ],
 
     // entry points
     control: {
-        taskGrid: {
-            competencyrowclick: 'onCompetencyRowClick'
+        courseSelector: {
+            select: 'onCourseSectionSelect'
+        },
+        dashboardCt: {
+            coursesectionselect: 'onDashboardSectionChange'
+        },
+        tasksGrid: {
+            cellclick: 'onTasksGridCellClick',
+            subcellclick: 'onTasksGridCellClick'
+        },
+        assignmentsComboField: {
+            render: 'onAssigneeComboRender'
+        },
+
+        'slate-tasks-teacher-taskrater button[action=accept]': {
+            click: 'onAcceptTaskClick'
+        },
+        'slate-tasks-teacher-taskrater button[action=reassign]': {
+            click: 'onAssignRevisionClick'
+        },
+        'slate-tasks-teacher-taskrater button[action=unassign]': {
+            click: 'onUnassignStudentTaskClick'
+        },
+        'slate-tasks-teacher-taskrater button[action=edit]': {
+            click: 'onEditStudentTaskClick'
+        },
+        'slate-tasks-teacher-taskrater slate-ratingview': {
+            rateskill: 'onRateSkillClick'
+        },
+        'slate-tasks-teacher-taskassigner button[action=assign]': {
+            click: 'onAssignTaskClick'
+        },
+        'slate-tasks-teacher-taskeditor button[action=save]': {
+            click: 'onSaveTaskClick'
+        },
+        'slate-tasks-teacher-appheader button[action=create]': {
+            click: 'onCreateTaskClick'
+        }
+    },
+
+    routes: {
+        'section/:sectionId': {
+            sectionId: '([a-zA-Z0-9])+',
+            action: 'viewCourseSection'
```

Comment:
@nbey our convention for naming route handles in `showCourseSection(sectionID)`. Please see https://github.com/JarvusInnovations/jarvus-guide-to-extjs/blob/master/applications/controllers.md